src/image/Image.cpp

```diff
@@ -4,15 +4,24 @@
 #include <pfm/pfm_input_file.hpp>
 #include <pic/pic_input_file.hpp>
 
+#include <QImage>
+
 namespace hdrv {
 
-Image::Image(int w, int h, int c, std::unique_ptr<float[]> data)
+Image::Image(int w, int h, int c, Format f, const std::vector<uint8_t>&& data)
```

Comment:
Removed the const. rvalue should prevent accidential copies.

I changed it to vector because I had problems with casting the unique_ptr between float and uint8_t. Should have worked, but I gave up and used vector instead. I don't think this is an issue regarding performance.